### PR TITLE
Move core container off of the testbot

### PR DESCRIPTION
**Pull request by rcooke-warwick**
This PR (WIP) is following a discussion earlier this week - 

- The core and worker containers currently sit on the testbot (in the case of the testbot worker)
- Tests run in the core container, the worker container acts as an API to the DUT and hardware
- This PR move the core container off of the testbot and on to the server (laptop/jenkins) to sit next to the client

Why?
- It will be easier to manage versions of core + tests
- Less complexity - tests no longer have to be sent to the testbot, logs don't have to be streamed back, we're more resistant to intermitant connectivity drops between client/testbot

What else had to change:
- New http endpoints added to the worker container, and worker helper functions updated to take those into account. Mainly these new endpoints are to allow SSH access to the DUT over the local testbot/dut network from the tests running remotely, allowing local pushing of containers.
- Had to move the status and start endpoints to the worker instead of core, to allow the client to find an idle worker + signify if a worker is in use. 

Todo:
- make screen capturing work
- tests may need to be updated
- simplify code in core / client that was used to send tests + image to testbot in the past

### CODE REVIEW

File: docker-compose.client.yml [hunk 1]
```diff
@@ -7,3 +7,12 @@ services:
       - "${WORKSPACE:-./workspace}:/usr/src/app/workspace:ro"
       - "${REPORTS:-./workspace/reports}:/usr/src/app/reports:rw"
       - "${SUITES:-./suites}:/usr/src/app/suites:ro"
+  core:
+    build: ./core
+    network_mode: 'host'
```
Comment (klutchell) [inline docker-compose.client.yml:12]:
```suggestion
    ports:
      - 80:80
```
No need for host network anymore.

File: worker/lib/index.ts [hunk 2]
```diff
@@ -280,6 +291,119 @@ async function setup(): Promise<express.Application> {
...
+					}
+			} catch (err) {
+				console.error(err);
+				next(err);
+			}
+		},
+	);
+
+	app.post(
+		`/exec`,
```
Comment (klutchell) [inline worker/lib/index.ts:467]: This is to exec a command on the worker? Can we get some comments?

File: worker/lib/index.ts [hunk 3]
```diff
@@ -280,6 +291,119 @@ async function setup(): Promise<express.Application> {
...
+				res.send('OK');
+			} catch (err) {
+				console.error(err);
+				next(err);
+			}
+		},
+	);
+
+	app.post(
+		'/dut/exec',
```
Comment (klutchell) [inline worker/lib/index.ts:336]: This is to exec a command on the DUT? Can we get some comments?

File: worker/lib/index.ts (hunk 2 above)
Comment (vipulgupta2048) [reply worker/lib/index.ts:467]: @rcooke-warwick Maybe transfer the comments from the helpers.

Comment (rcooke-warwick) [reply worker/lib/index.ts:467]: Thats right, this is to exec a command on the worker/testbot. I was 50/50 on whether we should allow people to that as it's un-contrained, but it's a way of getting the worker to do some things without more endpoints - e.g the stuff we do in the bluetooth test right now.

File: worker/lib/index.ts (hunk 3 above)
Comment (rcooke-warwick) [reply worker/lib/index.ts:336]: Correct, this is to execute a command in the DUT host OS - will add comments

**Comment by rcooke-warwick** [comment]
@balena-ci retest

**Comment by rcooke-warwick** [comment]
@balena-ci rebase

**Comment by rcooke-warwick** [comment]
@balena-ci rebase

**Comment by rcooke-warwick** [comment]
@balena-ci rebase

**Comment by jakogut** [comment]
@rcooke-warwick Still looking at this PR, but there's a bunch of duplicated commits. Could you remove those?

**Comment by rcooke-warwick** [comment]
@balena-ci rebase

**Comment by rcooke-warwick** [comment]
Tomorrow I'm going to nuke this PR and make a new one with the changes we talked about. Reviewing and discussion can happen there

**Comment by klutchell** [comment]
Superseded by https://github.com/balena-os/leviathan/pull/650

**Pull request closed without merging**